File: js/converter.js
importPackage(javax.swing);
importPackage(java.awt);

var frame = new JFrame("Celsius to Fahrenheit Converter");
frame.defaultCloseOperation = JFrame.EXIT_ON_CLOSE;

var celsius = new JTextField("0");
celsius.columns = 5;

var fahrenheit = new JTextField("32");
fahrenheit.columns = 5;

var panel = new JPanel;
panel.add(celsius);
panel.add(new JLabel(" Celsius = "));
panel.add(fahrenheit);
panel.add(new JLabel(" Fahrenheit "));

frame.contentPane.add(panel, BorderLayout.CENTER);

var button = new JButton("Convert");

button.addActionListener(function(e) {
    var c, f;

    c = celsius.text;
    f = c * 9 / 5 + 32;
    return fahrenheit.text = f;
});

frame.contentPane.add(button, BorderLayout.SOUTH);
frame.pack();
frame.visible = true;

File: scala/converter.scala
import swing._
import event._

object Converter extends SimpleSwingApplication {
  def top = new MainFrame {
    val celsius = new TextField {
      text = "0"
      columns = 5
    }

    val fahrenheit = new TextField {
      text = "32"
      columns = 5
    }

    title = "Convert Celsius to Fahrenheit"
    contents = new BorderPanel {
      import BorderPanel.Position._

      val panel = new FlowPanel(celsius, new Label(" Celsius = "),
          fahrenheit, new Label(" Fahrenheit"))

      layout(panel) = North

      val button = new Button {
        text = "Convert"
      }

      layout(button) = South

      listenTo(button)
      reactions += {
        case ButtonClicked(`button`) =>
          val c = Integer.parseInt(celsius.text)
          val f = c * 9 / 5 + 32
          fahrenheit.text = Integer.toString(f)
      }
    }
  }
}

Converter.main(new Array[String](0))